modules/pricer/src/test/java/com/opengamma/strata/pricer/capfloor/VerticalSpreadVolatilityIborCapletFloorletBinaryPeriodPricerTest.java

```diff
@@ -0,0 +1,234 @@
+import com.opengamma.strata.collect.tuple.Pair;
+import com.opengamma.strata.market.param.CurrencyParameterSensitivities;
+import com.opengamma.strata.market.sensitivity.PointSensitivityBuilder;
+import com.opengamma.strata.pricer.rate.ImmutableRatesProvider;
+import com.opengamma.strata.product.capfloor.IborCapletFloorletBinaryPeriod;
+import com.opengamma.strata.product.capfloor.IborCapletFloorletPeriod;
+import com.opengamma.strata.product.rate.IborRateComputation;
+
+/**
+ * Test {@link VerticalSpreadVolatilityIborCapletFloorletBinaryPeriodPricer}.
+ */
+public class VerticalSpreadVolatilityIborCapletFloorletBinaryPeriodPricerTest {
+
+  private static final ReferenceData REF_DATA = ReferenceData.standard();
+  private static final ZonedDateTime VALUATION = dateUtc(2008, 8, 18);
+  private static final LocalDate FIXING = LocalDate.of(2011, 1, 3);
+  private static final double AMOUNT = 1000000; //1m
+  private static final double STRIKE = 0.01;
+
+  private static final IborRateComputation RATE_COMP = IborRateComputation.of(EUR_EURIBOR_3M, FIXING, REF_DATA);
+  private static final IborCapletFloorletBinaryPeriod CAPLET_LONG = IborCapletFloorletBinaryPeriod.builder()
+      .caplet(STRIKE)
+      .startDate(RATE_COMP.getEffectiveDate())
+      .endDate(RATE_COMP.getMaturityDate())
+      .yearFraction(RATE_COMP.getYearFraction())
+      .amount(AMOUNT)
+      .iborRate(RATE_COMP)
+      .build();
+  private static final IborCapletFloorletBinaryPeriod CAPLET_SHORT = IborCapletFloorletBinaryPeriod.builder()
+      .caplet(STRIKE)
+      .startDate(RATE_COMP.getEffectiveDate())
+      .endDate(RATE_COMP.getMaturityDate())
+      .yearFraction(RATE_COMP.getYearFraction())
+      .amount(-AMOUNT)
+      .iborRate(RATE_COMP)
+      .build();
+  private static final IborCapletFloorletBinaryPeriod FLOORLET_LONG = IborCapletFloorletBinaryPeriod.builder()
+      .floorlet(STRIKE)
+      .startDate(RATE_COMP.getEffectiveDate())
+      .endDate(RATE_COMP.getMaturityDate())
+      .yearFraction(RATE_COMP.getYearFraction())
+      .amount(AMOUNT)
+      .iborRate(RATE_COMP)
+      .build();
+  private static final IborCapletFloorletBinaryPeriod FLOORLET_SHORT = IborCapletFloorletBinaryPeriod.builder()
+      .floorlet(STRIKE)
+      .startDate(RATE_COMP.getEffectiveDate())
+      .endDate(RATE_COMP.getMaturityDate())
+      .yearFraction(RATE_COMP.getYearFraction())
+      .amount(-AMOUNT)
+      .iborRate(RATE_COMP)
+      .build();
+
+  // valuation date before fixing date
+  private static final ImmutableRatesProvider RATES = IborCapletFloorletSabrRateVolatilityDataSet.getRatesProvider(
+      VALUATION.toLocalDate(), EUR_EURIBOR_3M, LocalDateDoubleTimeSeries.empty());
+  private static final SabrParametersIborCapletFloorletVolatilities VOLS = IborCapletFloorletSabrRateVolatilityDataSet
+      .getVolatilities(VALUATION, EUR_EURIBOR_3M);
+  // valuation date after fixing date
+  // valuation date equal to fixing date
```

Comment:
Looks like this comment can be removed as the rates provider below is for val date after fixing date?

Reply:
Yes, copy and paste from another test. The fixing date issue is dealt (and tested) with at the "VolatilityIborCapletFloorletPeriodPricer" level. I remove the comments.